src/main/java/jp/co/future/uroborosql/AbstractAgent.java

```diff
@@ -73,6 +80,9 @@
 	/** 例外発生にロールバックが必要なDBでリトライを実現するために設定するSavepointの名前 */
 	protected static final String RETRY_SAVEPOINT_NAME = "__retry_savepoint";
 
+	/** 一括更新用のバッチフレームの判定条件 */
+	private static final BiPredicate<Integer, Object> DEFAULT_BATCH_WHEN_CONDITION = (count, row) -> count == 1000;
```

Comment:
Is it not necessary to pass SqlContext or Entity of the previous line to BatchWhenCondition?

Usage example: When you want to insert when the key value of the line changes
Store the value of the key of the previous row in SqlContext # contextAttrs () and compare it with the value of this row

Reply:
In the case of `BULK INSERT`, it is not possible to create `SqlContext` at the judgment timing.
If you want to compare keys, we need to think about another way.

Reply:
I did not know that SqlContext can set SQL after generating SqlContext.
I will to fix this.